src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,141 @@
+        mapping field names to AttributesColumn objects.
+    trans_field_func - A function to transform the field name to valid column name in the filter query
+        before applying it to the FilterSet. Default is None.
+
+    """
+    for field, querystring in filter_query.items():
+        if field not in columns:
+            raise errors.IllegalParameterError(f"No such filter field: {field}")
+        column = columns[field]
+        minlen = analyzers.get_minimum_query_length(column.filter_strategy)
+        if minlen and len(querystring) < minlen:
+            raise errors.IllegalParameterError(
+                f"Filter field '{field}' requires a minimum query length of {minlen}")
+
+        # Translates the query field name to an available column name if necessary.
+        # For example, translates query field name '1' to a valid column name 'col_1_val'.
+        field = trans_field_func(field) if trans_field_func else field
+
+        fs.append(
+            field,
+            column.type,
+            querystring,
+            analyzers.get_analyzer(column.filter_strategy),
+            column.filter_strategy
+        )
+    return fs
+
+
+async def get_filters(
+        r: Request,
+        arango_coll: str,
+        coll_id: str,
+        load_ver: str,
+        load_ver_override: bool,
+        data_product: str,
+        get_columns_func: callable,
+        view_name: str = None,
+        count: bool = False,
+        sort_on: str = None,
+        sort_desc: bool = False,
+        filter_conjunction: bool = True,
+        match_spec: SubsetSpecification = None,
+        selection_spec: SubsetSpecification = None,
+        keep: dict[str, set[col_models.ColumnType]] = None,
+        keep_filter_nulls: bool = False,
+        skip: int = 0,
+        limit: int = 1000,
+        start_after: str = None,
+        trans_field_func: callable = None,
+) -> FilterSet:
+    """
+    Constructs a FilterSet and applies filters using the provided filter query and columns.
+
+    Requires a 'get_columns_func' to generate columns, represented as a dictionary mapping column name/ID to
+    columnar_attribs_common_models.AttributesColumn objects.
+    """
+    filter_query = _get_filter_map(r)
+    if filter_query and not view_name:
+        if load_ver_override:
+            # If we need this feature than the admin needs to supply the view name to use
+            # via the API
+            raise ValueError("Filtering is not supported with a load version override.")
+        raise ValueError(f"No search view name configured for collection {coll_id}, "
+                         + f"data product {data_product}. Cannot perform filtering operation")
+
+    # The caller is responsible for ensuring that each column retrieved through 'get_columns_func'
+    # is of type 'columnar_attribs_common_models.AttributesColumn'
+    columns = await get_columns_func(r, coll_id, load_ver, load_ver_override)
```

Comment:
Looking at this now I'm trying to figure out why we need to pass in a function at all. Is there any reason we can't just pass in the ColumnAttributesMeta or the list of columns and make things quite a bit simpler?

Reply:
Not for heatmap. We don't have a list of columns for heatmap.

Reply:
What will this function return for heatmaps?

Reply:
So why can't we just pass that in rather than a function to get it?

And if we can we should just pass the list of columns and have the get_filters turn it into a map

Reply:
yea. we can. Just to be clear, you are saying move what's in `_get_genome_attribs_columns` to the `get_genome_attributes` function? And having get_filters to take `columns` maybe a `columns` list?

Reply:
Well, you could leave _ggac alone and just call it and pass the result to get_filters as a list I think

Reply:
👍

Reply:
Looking at this with hindsight, `_get_genome_attribs_columns` is now just a single instruction - it seems like it could just be inlined. Like this maybe

```
    columns = await _get_genome_attribs_columns(appstate.arangostorage, collection_id, load_ver, load_ver_override)
    filters = await get_filters(
        r,
        names.COLL_GENOME_ATTRIBS,
        collection_id,
        load_ver,
        load_ver_override,
        ID,
        (await _get_genome_attributes_meta_internal(
            storage, coll_id, load_ver, load_ver_override)).columns,
        view_name=coll.get_data_product(ID).search_view if coll else None,
...
```

Reply:
👍

Reply:
:+1: